baseapp/abci_utils.go

```diff
@@ -394,11 +394,15 @@ func (ts *defaultTxSelector) SelectTxForProposal(_ context.Context, maxTxBytes,
 				ts.totalTxGas += txGasLimit
 				ts.totalTxBytes += txSize
 				ts.selectedTxs = append(ts.selectedTxs, txBz)
+			} else {
+				return true
```

Comment:
How does this fix not selecting the correct sequenced tx? In your example, won't this still select tx3?

Reply:
‘’‘
stop := h.txSelector.SelectTxForProposal(ctx, uint64(req.MaxTxBytes), maxBlockGas, memTx, txBz)
if stop {
	break
}
’‘’
stop=true will break for loop,so wont choose tx3
ur previous func will return [tx1,tx3](cause non-sequential sequence)，[our team often encounters this problem when the block is full]
so i provide two way to fix this bug，
first one is this pr，just skip all remaining txs (our team choose it)
second  is another pr（https://github.com/cosmos/cosmos-sdk/pull/19119），just skip non-sequential sequence sender，not every one

so can u give me a reply which way do you think is more reasonable? let me change that pr.
Thank you very much

Reply:
> How does this fix not selecting the correct sequenced tx? In your example, won't this still select tx3?

I later understand what you mean
I missed the test about maxgas，i can add a test case
but actually I still want to know which solution you think is better